<?xml version="1.0" encoding="UTF-8"?> 
<launch> 
    <param name="/use_sim_time" value="true"/>

    <node name="republish_left" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera/infra1/image_rect_raw raw out:=/camera/left/image_raw" /> 
    <node name="republish_right" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera/infra2/image_rect_raw raw out:=/camera/right/image_raw" />

    <arg name="vocabulary" default="/root/catkin_ws/src/ORB_SLAM3/Vocabulary/ORBvoc.txt" doc="Vocabulary for DBoW2."/> 
    <arg name="settings" default="/root/catkin_ws/src/ORB_SLAM3/Examples/ROS/ORB_SLAM3/config/agro_sim_stereo_inertial.yaml" doc="System settings."/> 
    <arg name="map_frame_id" default="map"/>
    <arg name="base_link_frame_id" default="base_link_vis"/>
    <arg name="do_rectify" default="false" />
    <node name="orbslam3_stereo_inertial" pkg="ORB_SLAM3" type="Stereo" output="screen" args="$(arg vocabulary) $(arg settings) $(arg do_rectify)"> 
        <!-- <remap from="/camera/left/image_raw" to="/camera/infra1/image_rect_raw"/>
        <remap from="/camera/right/image_raw" to="/camera/infra2/image_rect_raw"/>            -->
        <!-- <remap from="/imu" to="/camera/imu/sample"/>-->
        <param name="map_frame_id" value="$(arg map_frame_id)"/>
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)"/>
    </node>

    <!-- <node name="image_converter_l" type="image_proc" pkg="image_proc" ns="/camera/infra1" output="screen" >
        <remap from="image_raw" to="image_rect_raw" />
    </node>

    <node name="image_converter_r" type="image_proc" pkg="image_proc" ns="/camera/infra2" output="screen" >
         <remap from="image_raw" to="image_rect_raw" />
    </node> -->


     <node pkg="tf2_ros" type="static_transform_publisher" name="start_broadcaster" args="0 0 0 0 0 0 start map"/>
</launch>
